hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -313,6 +328,10 @@ public Long countTargetByFilters(final Collection<TargetUpdateStatus> status, fi
             specList.add(TargetSpecifications.isOverdue(TimestampCalculator.calculateOverdueTimestamp()));
         }
         if (filterParams.getFilterByDistributionId() != null) {
+            if (!distributionSetRepository.exists(filterParams.getFilterByDistributionId())) {
```

Comment:
I think you can call the method `throwEntityNotFoundIfDsDoesNotExist` from line 258 here, is the same exists check, right?